File: sdk/ios/Sources/InviteSDK/Core/Logger.swift
// Copyright 2024 share-installs Authors.

import Foundation
import os.log

/// Internal SDK logger. Uses os.log for performance and privacy compliance.
enum InviteLogger {
    private static let subsystem = "com.invitesdk"

    private static let general = Logger(subsystem: subsystem, category: "General")
    private static let network = Logger(subsystem: subsystem, category: "Network")
    private static let deepLink = Logger(subsystem: subsystem, category: "DeepLink")
    private static let fingerprint = Logger(subsystem: subsystem, category: "Fingerprint")

    static var isDebugEnabled = false

    static func debug(_ message: String, category: LogCategory = .general) {
        guard isDebugEnabled else { return }
        logger(for: category).debug("[InviteSDK] \(message, privacy: .public)")
    }

    static func info(_ message: String, category: LogCategory = .general) {
        logger(for: category).info("[InviteSDK] \(message, privacy: .public)")
    }

    static func warning(_ message: String, category: LogCategory = .general) {
        logger(for: category).warning("[InviteSDK] \(message, privacy: .public)")
    }

    static func error(_ message: String, category: LogCategory = .general) {
        logger(for: category).error("[InviteSDK] \(message, privacy: .public)")
    }

    static func fault(_ message: String, category: LogCategory = .general) {
        logger(for: category).fault("[InviteSDK] \(message, privacy: .public)")
    }

    private static func logger(for category: LogCategory) -> Logger {
        switch category {
        case .general: return general
        case .network: return network
        case .deepLink: return deepLink
        case .fingerprint: return fingerprint
        }
    }

    enum LogCategory {
        case general, network, deepLink, fingerprint
    }
}

File: sdk/android/invitesdk/src/main/kotlin/com/invitesdk/core/Logger.kt
/*
 * Copyright 2024 share-installs Authors.
 */

package com.invitesdk.core

import android.util.Log

/** Internal logger for the InviteSDK. */
internal object InviteLogger {

    private const val TAG = "InviteSDK"
    var isDebugEnabled: Boolean = false

    fun d(message: String) {
        if (isDebugEnabled) Log.d(TAG, message)
    }

    fun i(message: String) {
        Log.i(TAG, message)
    }

    fun w(message: String, throwable: Throwable? = null) {
        if (throwable != null) Log.w(TAG, message, throwable)
        else Log.w(TAG, message)
    }

    fun e(message: String, throwable: Throwable? = null) {
        if (throwable != null) Log.e(TAG, message, throwable)
        else Log.e(TAG, message)
    }
}
